### DISCUSSION TITLE
Per-project(/dir) app version configuration?

### DISCUSSION THREAD

**Discussion by crispinb** [ScoopInstaller/Scoop#4695] [Ideas]
Given scoop can install multiple versions of an app, is there any potential for an enhancement allowing a version to be specified for a specific folder/dir? I'm thinking of the way asdf uses '.tool-versions' for per-project version config. 

I admit I'm a bit in the dark about how scoop works (I see there are shims but they seem not to be used for launching apps?), so it's possible I'm asking a daft question (sorry if so!).

**Comment by HUMORCE** [comment]
https://github.com/ScoopInstaller/Scoop/wiki/Switching-Ruby,-Python-and-PHP-Versions

if you do not want always switching them, then use absolute paths in tools you are using.

or if you want to switch to 'patch' version, example:

```
> python --version
Python 3.10.2
> scoop info python
Name: python
Description: A programming language that lets you work quickly and integrate systems more effectively.
Version: 3.10.2
Website: https://www.python.org/
License: Python-2.0 (https://spdx.org/licenses/Python-2.0.html)
Manifest:
  C:\Users\HUMOR\scoop\buckets\main\bucket\python.json
Installed:
  C:\Users\HUMOR\scoop\apps\python\3.10.1
  C:\Users\HUMOR\scoop\apps\python\3.10.2
> scoop reset python@3.10.1
> python --version
Python 3.10.1
```

maybe you need this too: `scoop info main/direnv`

**Reply by crispinb** [reply]
Yep I had found `scoop reset`, but I was wondering if the scoop architecture would permit an enhancement to avoid the manual step. The parallel I'm thinking of is asdf https://github.com/asdf-vm/asdf. It has a `local` command which allows you to set a version for a particular directory (via adding a ./.tool-versions file)

**Reply by HUMORCE** [reply]
read @rashil2000 ‘s comments

**Reply by goyalyashpal** [reply]
> maybe you need this too: `scoop info main/direnv`

yeah, this was my first though on reading this request

**Comment by rashil2000** [comment]
Scoop does not have such a feature, and it's unlikely that it'll ever do. This is because it's a generic package manager. Such facility makes sense for individual toolchain managers like nvm or asdf, some of which you can install through Scoop itself.

**Reply by airtonix** [reply]
asdf doesn't work on windows though.

**Comment by rashil2000** [comment]
Additionally, shortcuts are used to access GUI programs, and shims are used for console/terminal programs.

**Reply by crispinb** [reply]
Sorry didn't see your reply before writing the above. asdf is also pretty generic. But I take your point about scoop's purpose.
